### DISCUSSION TITLE
Dynamically assign role with Public_Registration

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#11117] [Archive]
Hi,

I have 2 differents front applications that use the same default oauth flow in the same Directus instance.
The public registration is enabled. I saw we can with an environnment variable assign a static role.

How could I assign a different role based on the application ?

I cannot find yet any workaround using hooks,

_myapp1_ would assign by default 'App1Role' when the user is created
_myapp2_ would assign by default 'App2Role' when the user is created

EDIT: User must be able to authenticate after in one or the other app.

Thanks for help

**Comment by joselcvarela** [comment]
~~I think you may need to use custom API hooks here.~~
~~https://docs.directus.io/extensions/hooks/#custom-api-hooks~~

See @aidenfoxx answer https://github.com/directus/directus/discussions/11117#discussioncomment-1986444

**Comment by aidenfoxx** [comment]
Alternatively you could also configure 2 seperate instances of the same oauth provider, just with different static roles e.g. `AUTH_PROVIDERS="google_app1,google_app2"`?

**Reply by joselcvarela** [reply]
Totally forgot about this variable  `AUTH_<PROVIDER>_DEFAULT_ROLE_ID`.
Nice one 👍

**Reply by gkielwasser** [reply]
> Alternatively you could also configure 2 seperate instances of the same oauth provider, just with different static roles e.g. `AUTH_PROVIDERS="google_app1,google_app2"`?

This is working so well, Thank you @aidenfoxx !!

**Reply by gkielwasser** [reply]
@aidenfoxx It seems I just notice a blocker with this solution but I'm not sure. 
Is is intended that with 2 separates google instance, a user registered with `google_app1` cannot login into `google_app2` (ofc also the opposite)? Even if both google providers are configured the same way ? It looks like the provider name is checked ?

In my case whatever from where the user were registering, he could login into the other app.

**Reply by joselcvarela** [reply]
I believe the `provider` field on `directus_users` is causing this.
I am not sure this have been made on purpose or if it was not taken into account.
Let's wait for Aiden to see what are his thoughts 👍

**Reply by aidenfoxx** [reply]
This is intended behavior. You can see https://github.com/directus/directus/issues/9331 for a discussion about this.

The idea is that a user on a managed provider like LDAP shouldn't be able to bypass those auth protocols by logging in with Facebook, for example. It would defeat the point of a managed auth provider. So by default users are limited to whatever auth provider they are created with.

It's still in discussion because it's hard to know what is the best solution.

**Reply by gkielwasser** [reply]
Ok I get it.

So I'm still seeking a workaround for my initial usecase without restricting my user to be only able to connect on the app they have registered. I will update the discussion.
